src/React.re

```diff
@@ -0,0 +1,262 @@
+      "fallback": option(element),
+      "maxDuration": option(int),
+    } =
+    "";
+  [@bs.module "react"]
+  external make:
+    component({
+      .
+      "children": option(element),
+      "fallback": option(element),
+      "maxDuration": option(int),
+    }) =
+    "Suspense";
+};
+
+/* HOOKS */
+
+/*
+ * Yeah, we know this api isn't great. tl;dr: useReducer instead.
+ * It's because useState can take functions or non-function values and treats
+ * them differently. Lazy initializer + callback which returns state is the
+ * only way to safely have any type of state and be able to update it correctly.
+ */
+[@bs.module "react"]
+external useState:
+  ([@bs.uncurry] (unit => 'state)) => ('state, ('state => 'state) => unit) =
+  "";
+
+[@bs.module "react"]
+external useReducer:
+  ([@bs.uncurry] (('state, 'action) => 'state), 'state) =>
+  ('state, 'action => unit) =
+  "";
+
+[@bs.module "react"]
+external useReducerWithMapState:
+  (
+    [@bs.uncurry] (('state, 'action) => 'state),
+    'initialState,
+    'initialState => 'state
+  ) =>
+  ('state, 'action => unit) =
+  "useReducer";
+
+[@bs.module "react"]
+external useEffect: ([@bs.uncurry] (unit => option(unit => unit))) => unit =
+  "useEffect";
+[@bs.module "react"]
+external useEffect0:
+  ([@bs.uncurry] (unit => option(unit => unit)), [@bs.as {json|[]|json}] _) =>
+  unit =
+  "useEffect";
+[@bs.module "react"]
+external useEffect1:
+  ([@bs.uncurry] (unit => option(unit => unit)), array('a)) => unit =
+  "useEffect";
+[@bs.module "react"]
+external useEffect2:
+  ([@bs.uncurry] (unit => option(unit => unit)), ('a, 'b)) => unit =
+  "useEffect";
+[@bs.module "react"]
+external useEffect3: (unit => option(unit => unit), ('a, 'b, 'c)) => unit =
+  "useEffect";
+[@bs.module "react"]
+external useEffect4: (unit => option(unit => unit), ('a, 'b, 'c, 'd)) => unit =
+  "useEffect";
+
+[@bs.module "react"]
+external useLayoutEffect:
+  ([@bs.uncurry] (unit => option(unit => unit))) => unit =
+  "useLayoutEffect";
+[@bs.module "react"]
+external useLayoutEffect0:
+  ([@bs.uncurry] (unit => option(unit => unit)), [@bs.as {json|[]|json}] _) =>
+  unit =
+  "useLayoutEffect";
+[@bs.module "react"]
+external useLayoutEffect1:
+  ([@bs.uncurry] (unit => option(unit => unit)), array('a)) => unit =
+  "useLayoutEffect";
+[@bs.module "react"]
+external useLayoutEffect2:
+  ([@bs.uncurry] (unit => option(unit => unit)), ('a, 'b)) => unit =
+  "useLayoutEffect";
+
+[@bs.module "react"]
+external useMemo: ([@bs.uncurry] (unit => 'any)) => 'any = "useMemo";
+[@bs.module "react"]
+external useMemo0:
+  ([@bs.uncurry] (unit => 'any), [@bs.as {json|[]|json}] _) => 'any =
+  "useMemo";
+[@bs.module "react"]
+external useMemo1: ([@bs.uncurry] (unit => 'any), array('a)) => 'any =
+  "useMemo";
+[@bs.module "react"]
+external useMemo2: ([@bs.uncurry] (unit => 'any), ('a, 'b)) => 'any =
+  "useMemo";
+[@bs.module "react"]
+external useMemo3: ([@bs.uncurry] (unit => 'any), ('a, 'b, 'c)) => 't =
```

Comment:
Whats ups with this type definition. The callback returns 'any, but return type is an arbitrary 't ?

Reply:
Updated! Thanks for pointing out.